simphony/io/tests/test_file_mesh.py

```diff
@@ -375,6 +375,93 @@ def test_get_all_cells_iterator(self):
 
         self.assertItemsEqual(icells_id, cuids)
 
+    def test_get_all_edges_iterator_with_data(self):
+        """ Checks the edge iterator
+
+        Checks that an interator over all
+        the edges of the mesh is returned
+        when the function iter_edges is called
+        without arguments and its data is correctly retrieved
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:3]]
+
+        edges = [
+            Edge(puids[0:2], data=DataContainer({CUBA.VELOCITY: [0, 0, 0]}))
+        ]
+
+        euids = [self.mesh.add_edge(edge) for edge in edges]
+
+        iedges = self.mesh.iter_edges()
+
+        iedges_id = [edge.uid for edge in iedges]
+        first_edge = self.mesh.iter_edges().next()
+
+        self.assertItemsEqual(iedges_id, euids)
+        self.assertIsNot(len(first_edge.data), 0)
```

Comment:
the correct link is https://github.com/simphony/simphony-common/blob/master/simphony/testing/abc_container_check.py#L15